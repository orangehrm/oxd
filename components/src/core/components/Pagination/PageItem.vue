<template>
  <li>
    <button type="button" :class="classes">
      <oxd-icon v-if="previous" name="chevron-left" />
      <oxd-icon v-else-if="next" name="chevron-right" />
      <template v-else>{{ page }}</template>
    </button>
  </li>
</template>

<script lang="ts">
import {defineComponent} from 'vue';
import Icon from '@orangehrm/oxd/core/components/Icon/Icon.vue';
import {pageableMixin} from '../../../mixins/pageable';

export default defineComponent({
  name: 'oxd-pagination-page-item',

  components: {
    'oxd-icon': Icon,
  },

  mixins: [pageableMixin],

  props: {
    previous: {
      type: Boolean,
    },
    next: {
      type: Boolean,
    },
    selected: {
      type: Boolean,
    },
  },

  computed: {
    classes(): object {
      return {
        'oxd-pagination-page-item': true,
        'oxd-pagination-page-item--page': !(this.previous || this.next),
        'oxd-pagination-page-item--page-selected': this.selected,
        'oxd-pagination-page-item--previous-next': this.previous || this.next,
      };
    },
  },
});
</script>

<style src="./page-item.scss" lang="scss" scoped></style>
