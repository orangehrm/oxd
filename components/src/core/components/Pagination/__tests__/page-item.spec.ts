import {mount} from '@vue/test-utils';
import PageItem from '@orangehrm/oxd/core/components/Pagination/PageItem.vue';

describe('PageItem.vue', () => {
  it('renders OXD PageItem', () => {
    const wrapper = mount(PageItem, {
      props: {page: 1},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders OXD PageItem previous', () => {
    const wrapper = mount(PageItem, {
      props: {previous: true},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders OXD PageItem next', () => {
    const wrapper = mount(PageItem, {
      props: {next: true},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders OXD PageItem selected', () => {
    const wrapper = mount(PageItem, {
      props: {page: 2, selected: true},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
