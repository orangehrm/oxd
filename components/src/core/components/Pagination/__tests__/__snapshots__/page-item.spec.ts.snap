// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageItem.vue renders OXD PageItem 1`] = `<li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">1</button></li>`;

exports[`PageItem.vue renders OXD PageItem next 1`] = `<li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-right"></i></button></li>`;

exports[`PageItem.vue renders OXD PageItem previous 1`] = `<li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-left"></i></button></li>`;

exports[`PageItem.vue renders OXD PageItem selected 1`] = `<li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">2</button></li>`;
